src/frontend/Tactics.ml

```diff
@@ -379,3 +378,39 @@ struct
     let+ trefl = RM.quote_con refl_tp refl in
     (trefl, refl_tp)
 end
+
+module Univ =
+struct
+  let as_code = function
+    | D.ElStable code -> RM.ret @@ D.StableCode code
+    | D.ElUnstable code -> RM.ret @@ D.UnstableCode code
+    | D.ElCut cut -> RM.ret @@ D.Cut { tp = D.Univ; cut }
+    | tp -> RM.expected_connective `El tp
+
+  let hcom_chk tac_src tac_trg tac_tm =
```

Comment:
The intention of the distinction between the `Refiner` and `Tactic` modules is that the former hosts trusted rules (ones that manually produce terms) and the latter hosts only things that are built up by combining existing trusted rules. As such I think there are two options:

1. The `hcom_chk` can be moved to the refiner.
2. The `hcom_chk` could be (potentially?) rewritten as an actual tactic, meaning something that doesn't produce terms itself but instead does so by calling the existing refinement rules. 

I'm not sure if the latter is possible, so I leave it to your judgment.

Reply:
Done